### PR TITLE
Test TLS password - [MOD-6119]

### PR DESCRIPTION
**Pull request by GuyAv46**
**Describe the changes in the pull request**

Remove the configuration of `tls` and add a designated test for TLS (and improve another test to run with TLS)

introducing `get_TLS_args()` in `common.py` that will return all the needed parameters to start an environment with TLS

**Mark if applicable**

- [ ] This PR introduces API changes
- [ ] This PR introduces serialization changes

### CODE REVIEW

**Review by alonre24 — APPROVED** [review]

File: tests/pytests/test_issues.py [hunk 1]
```diff
@@ -857,46 +857,54 @@ def mod5778_add_new_shard_to_cluster(env):
...
+                     'ssl_ca_certs': env.envRunner.shards[0].getTLSCACertFile(),
+                     'ssl_password': env.envRunner.tlsPassphrase})
+    with TimeLimit(10, 'waiting for new shard to acknowledge the topology change'):
+      while True:
+        time.sleep(0.05)
+        try:
+          new_instance_conn = RedisCluster(**kwargs)
+          break
+        except:
+          pass
```
Comment (alonre24) [inline tests/pytests/test_issues.py:876]: Did you understand why did it take longer for the new shard to upload after your changes? Or it is not related?

Comment (GuyAv46) [reply tests/pytests/test_issues.py:876]: No. Here it was just flaky, replacing arbitrary `sleep(5)` with a timed-out check.
Why was it taking 5 seconds before?

**Review by alonre24 — APPROVED** [review]

File: tests/pytests/test_issues.py [hunk 2]
```diff
@@ -823,18 +823,26 @@ def test_mod5791(env):
 
 
 @skip(asan=True)
-def test_mod5778_add_new_shard_to_cluster(env):
-  mod5778_add_new_shard_to_cluster(env)
+def test_mod5778_add_new_shard_to_cluster():
+    if server_version_is_less_than('7.0.0'):  # cluster shards command is not supported for redis < 7
+        return
```
Comment (GuyAv46) [inline tests/pytests/test_issues.py:828]:
```suggestion
        raise SkipTest()
```

File: tests/pytests/test_issues.py [hunk 3]
```diff
@@ -823,18 +823,26 @@ def test_mod5791(env):
...
+    mod5778_add_new_shard_to_cluster(Env())
+
 
 @skip(asan=True)
 def test_mod5778_add_new_shard_to_cluster_TLS():
-  cert_file, key_file, ca_cert_file, passphrase = get_TLS_args()
-  env = Env(useTLS=True, tlsCertFile=cert_file, tlsKeyFile=key_file, tlsCaCertFile=ca_cert_file, tlsPassphrase=passphrase)
-  mod5778_add_new_shard_to_cluster(env)
+    if server_version_is_less_than('7.0.0'):  # cluster shards command is not supported for redis < 7
+        return
```
Comment (GuyAv46) [inline tests/pytests/test_issues.py:835]:
```suggestion
        raise SkipTest()
```

File: tests/pytests/test_issues.py [hunk 4]
```diff
@@ -866,46 +891,35 @@ def mod5778_add_new_shard_to_cluster(env: Env):
...
         while True:
-          res = command()
-          if res == expected:
+            time.sleep(iteration_wait_time)
+            try:
+                new_instance_conn = RedisCluster(**kwargs)
+                break
+            except exceptions.RedisClusterException and IndexError:
+                pass  # these two exceptions indicate that the new shard still waking up
+    new_instance_conn = RedisCluster(**kwargs)
```
Comment (GuyAv46) [inline tests/pytests/test_issues.py:903]: redundant. happens in the loop

**Review by alonre24 — DISMISSED** [review]

**Review by alonre24 — APPROVED** [review]

**Pull request merged**